### PR TITLE
feat: stress-ckb-test

### PR DESCRIPTION
**Pull request by pygman**

### CODE REVIEW

File: offchain-modules/packages/scripts/src/stress-ckb-test.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  burn,
+  lock,
+  prepareCkbAddresses,
+  prepareCkbPrivateKeys,
+  prepareEthWallets,
+  check,
+} from './utils/ckb_batch_test';
+
+// your send lock tx account privkey
+const ethPrivateKey = '0x719e94ec5d2ecef67b5878503ffd6e1e0e2fe7a52ddd55c436878cb4d52d376d';
```
Comment (mkxbl) [inline offchain-modules/packages/scripts/src/stress-ckb-test.ts:18]:
```suggestion
const ethPrivateKey = '';
```

File: offchain-modules/packages/scripts/src/stress-ckb-test.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+// your transfer ckb to recipients account privkey
+const ckbPrivateKey = '0xa6b8e0cbadda5c0d91cf82d1e8d8120b755aa06bc49030ca6e8392458c65fc80';
```
Comment (mkxbl) [inline offchain-modules/packages/scripts/src/stress-ckb-test.ts:20]:
```suggestion
const ckbPrivateKey = '';
```

File: offchain-modules/packages/scripts/src/stress-ckb-test.ts [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+const ethNodeUrl = 'http://127.0.0.1:8545';
+const ckbNodeUrl = 'http://127.0.0.1:8114';
+const ckbIndexerUrl = 'http://127.0.0.1:8116';
+const forceBridgeUrl = 'http://127.0.0.1:8080/force-bridge/api/v1';
+const ckbTokenAddress = '0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff';
+const xchainTokenAddress = '0xe9B447cA594cB87B8d912040c8981B9696541B82';
+
+const lockCkbAmount = '30000000000000';
+const burnCkbSudtAmount = '10000000000000';
```
Comment (mkxbl) [inline offchain-modules/packages/scripts/src/stress-ckb-test.ts:29]: miss sudt test

File: offchain-modules/packages/scripts/src/stress-ckb-test.ts [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+  logger.info('start stress lock test');
+  const stressPromise: PromiseLike<void>[] = [];
+  const lockPromise = stressLock(roundNumber, batchNumber, client, ckb, ckbPrivateKeys, recipients);
+  stressPromise.push(lockPromise);
+  await Promise.all(stressPromise);
+  if (bridgeDirection === 'both') {
+    const ckbAddress = ckb.utils.privateKeyToAddress(ckbPrivateKey, { prefix: AddressPrefix.Testnet });
+    await stressBurn(roundNumber, batchNumber, client, provider, ethWallets, ckbAddress);
+  }
```
Comment (mkxbl) [inline offchain-modules/packages/scripts/src/stress-ckb-test.ts:87]: should stress lock and burn concurrently

File: offchain-modules/packages/scripts/src/stress-ckb-test.ts [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+  logger.info(`stress test succeed!`);
+}
+
+async function stressLock(
+  roundNumber: number,
+  batchNumber: number,
```
Comment (mkxbl) [inline offchain-modules/packages/scripts/src/stress-ckb-test.ts:93]: unused `batchNumber`,  should batch send lock txes

File: offchain-modules/packages/scripts/src/stress-ckb-test.ts [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+  client: JSONRPCClient,
+  provider: ethers.providers.JsonRpcProvider,
+  ethWallets: Array<ethers.Wallet>,
+  recipient: string,
+  intervalMs = 0,
+) {
+  logger.info(`start stress burn test`);
+  const burnCkbSudtTxs = await burn(
```
Comment (mkxbl) [inline offchain-modules/packages/scripts/src/stress-ckb-test.ts:129]: should batch send burn txes

File: offchain-modules/packages/scripts/src/testnetDocker.ts [hunk 7]
```diff
@@ -40,17 +42,24 @@ async function generateConfig(
...
   baseConfig.eth.assetManagerContractAddress = assetManagerContractAddress;
   baseConfig.eth.safeMultisignContractAddress = safeAddress;
   baseConfig.ckb.deps = ckbDeps;
   baseConfig.ckb.ownerCellTypescript = ownerCellConfig.ownerCellTypescript;
+  baseConfig.ckb.omniLockAdminCellTypescript = omniLockCellConfig.adminCellTypescript;
   baseConfig.ckb.startBlockHeight = ckbStartHeight;
   baseConfig.eth.startBlockHeight = ethStartHeight;
+  baseConfig.eth.safeMultisignContractNetworks = contractNetworks;
+  baseConfig.eth.lockNervosAssetFee = '100000000';
+  baseConfig.eth.burnNervosAssetFee = '0';
```
Comment (mkxbl) [inline offchain-modules/packages/scripts/src/testnetDocker.ts:62]:
```suggestion
  baseConfig.eth.burnNervosAssetFee = '10000';
```

Comment (pygman) [reply offchain-modules/packages/scripts/src/testnetDocker.ts]: fixed

File: offchain-modules/packages/scripts/src/stress-ckb-test.ts (hunk 4 above)
Comment (pygman) [reply offchain-modules/packages/scripts/src/stress-ckb-test.ts]: fixed

File: offchain-modules/packages/scripts/src/stress-ckb-test.ts (hunk 6 above)
Comment (pygman) [reply offchain-modules/packages/scripts/src/stress-ckb-test.ts:174]: fixed

File: offchain-modules/packages/scripts/src/stress-ckb-test.ts (hunk 3 above)
Comment (pygman) [reply offchain-modules/packages/scripts/src/stress-ckb-test.ts]: fixed

File: offchain-modules/packages/scripts/src/stress-ckb-test.ts (hunk 5 above)
Comment (pygman) [reply offchain-modules/packages/scripts/src/stress-ckb-test.ts]: fixed

File: offchain-modules/packages/scripts/src/stress-ckb-test.ts (hunk 2 above)
Comment (pygman) [reply offchain-modules/packages/scripts/src/stress-ckb-test.ts]: removed

File: offchain-modules/packages/scripts/src/stress-ckb-test.ts (hunk 1 above)
Comment (pygman) [reply offchain-modules/packages/scripts/src/stress-ckb-test.ts]: removed

File: offchain-modules/packages/scripts/src/stress-schedule.ts [hunk 8]
```diff
@@ -0,0 +1,301 @@
...
+  sudtTypescriptHash: string;
+  sudtArgs: string;
+  xchainSudtTokenAddress: string;
+  lockCkbSudtAmount: string;
+  burnErc20SudtAmount: string;
+  sudtTypescript: ConfigItem;
+}
+
+async function stressTest() {
+  initLog({ level: 'debug', identity: 'stress-ckb-test', logFile: './data/stress-schedule.log' });
```
Comment (mkxbl) [inline offchain-modules/packages/scripts/src/stress-schedule.ts:194]:
```suggestion
  initLog({ level: 'debug', identity: 'stress-schedule-test', logFile: './data/stress-schedule.log' });
```

Comment (pygman) [reply offchain-modules/packages/scripts/src/stress-schedule.ts]: fixed

File: offchain-modules/packages/scripts/src/stress-ckb-test.ts [hunk 9]
```diff
@@ -0,0 +1,316 @@
...
+      intervalMs,
+    );
+    await check(client, 1, 'Nervos', sudtTypescriptHash, 'Ethereum', recipients, lockErc20Txs);
+    logger.info(`${i + 1} round stress ckb lock test succeed`);
+  }
+  logger.info(`stress ckb lock test succeed!`);
+}
+
+async function stressBurn(
+  roundNumber: number,
```
[the remark this answers is not in the document]
Comment (pygman) [reply offchain-modules/packages/scripts/src/stress-ckb-test.ts:261]: fixed

File: offchain-modules/packages/scripts/src/stress-ckb-test.ts [hunk 10]
```diff
@@ -0,0 +1,316 @@
...
+      client,
+      rpc,
+      ckbPrivateKeys,
+      ckbAddresses,
+      recipients,
+      sudtTypescriptHash,
+      (BigInt(lockCkbSudtAmount) * BigInt(multiple)).toString(10),
+      intervalMs,
+    );
+    await check(client, 1, 'Nervos', sudtTypescriptHash, 'Ethereum', recipients, lockErc20Txs);
```
[the remark this answers is not in the document]
Comment (pygman) [reply offchain-modules/packages/scripts/src/stress-ckb-test.ts]: fixed

File: offchain-modules/packages/scripts/src/stress-ckb-test.ts [hunk 11]
```diff
@@ -0,0 +1,316 @@
...
+      rpc,
+      ckbPrivateKeys,
+      ckbAddresses,
+      recipients,
+      CKB_TYPESCRIPT_HASH,
+      (BigInt(lockCkbAmount) * BigInt(multiple)).toString(10),
+      intervalMs,
+    );
+    await check(client, 1, 'Nervos', CKB_TYPESCRIPT_HASH, 'Ethereum', recipients, lockCkbTxs);
+    const lockErc20Txs = await lock(
```
[the remark this answers is not in the document]
Comment (pygman) [reply offchain-modules/packages/scripts/src/stress-ckb-test.ts]: renamed

File: offchain-modules/packages/scripts/src/stress-ckb-test.ts [hunk 12]
```diff
@@ -0,0 +1,316 @@
...
+      client,
+      rpc,
+      ckbPrivateKeys,
+      ckbAddresses,
+      recipients,
+      CKB_TYPESCRIPT_HASH,
+      (BigInt(lockCkbAmount) * BigInt(multiple)).toString(10),
+      intervalMs,
+    );
+    await check(client, 1, 'Nervos', CKB_TYPESCRIPT_HASH, 'Ethereum', recipients, lockCkbTxs);
```
[the remark this answers is not in the document]
Comment (pygman) [reply offchain-modules/packages/scripts/src/stress-ckb-test.ts]: fixed

File: offchain-modules/packages/x/src/config.ts [hunk 13]
```diff
@@ -69,6 +71,7 @@ export interface EthConfig {
   lockNervosAssetFee: string; // bridge fee paid by CKB, unit: shannon
   burnNervosAssetFee: string; // bridge fee paid by ETH, unit: wei
   assetManagerContractAddress: string;
+  networks?: ContractNetworksConfig;
```
[the remark this answers is not in the document]
Comment (pygman) [reply offchain-modules/packages/x/src/config.ts]: removed

File: offchain-modules/packages/x/src/core.ts [hunk 14]
```diff
@@ -124,6 +125,24 @@ export class ForceBridgeCore {
...
+        .map((asset) => {
+          if (!asset.typescriptHash) {
+            asset.typescriptHash = utils.computeScriptHash({
+              code_hash: typescript.codeHash,
+              hash_type: typescript.hashType,
+              args: asset.sudtArgs!,
+            });
+          }
+        });
+    }
```
[the remark this answers is not in the document]
Comment (pygman) [reply offchain-modules/packages/x/src/core.ts]: fixed

**Pull request merged**